### DISCUSSION TITLE
How to fix full-width and full-height content

### DISCUSSION THREAD

**Discussion by surushTodzhibekov** [quasarframework/quasar#11081] [General - Components / Directives / etc] [answered]
Hi guys! I've issues with full-width after adding texts, it displays in one line. I appreciate if someone help me.

https://user-images.githubusercontent.com/38825951/138519771-2d072f1d-df92-4141-82b5-e9feb8f635d5.mp4

<template>
  <q-page class="relative-position">
    <q-scroll-area class="absolute full-height full-width" >
    <div class="q-py-lg q-px-md row items-end q-col-gutter-md">
      <div class="col">
        <q-input
          bottom-slots
          class="new-qweet"
          v-model="newQweetContent"
          placeholder="What's happening? "
          counter
          maxlength="282"
          autogrow
        >
          <template v-slot:before>
            <q-avatar size="lg">
              <img src="https://cdn.quasar.dev/img/avatar5.jpg">
            </q-avatar>
          </template>
        </q-input>
      </div>
      <div class="col col-shrink">
        <q-btn
          @click="addNewTweet"
          :disable="!newQweetContent"
          unelevated
          rounded
          no-caps
          color="primary"
          label="Tweet"
          class="q-mb-lg"
        />
      </div>
    </div>
    <q-separator
      size="10px"
      color="grey-2"
      class="divider"
    />

    <q-list separator>
      <transition-group
        appear
        enter-active-class="animated fadeIn slower"
        leave-active-class="animated fadeOut slower"
      >
      <q-item v-for="qweet in qweets" :key="qweet.id" class="qweet q-py-md">
        <q-item-section avatar top>
          <q-avatar size="lg">
            <img src="https://cdn.quasar.dev/img/avatar2.jpg">
          </q-avatar>
        </q-item-section>

        <q-item-section>
          <q-item-label class="text-subtitle1">
            <strong>Todzhibekov Surush</strong>
            <span class="text-grey-7">
              @surushAmrobek
              <br class="lt-md">&bull;  {{ relativeDate(qweet.date) }}
            </span>
          </q-item-label>
          <q-item-label class="qweet-content text-body1">
           {{qweet.content}}
          </q-item-label>
          <div class="qweet-icons row justify-between q-mt-sm">
            <q-btn
              flat
              round
              color="grey"
              icon="fas fa-comment"
              size="sm"
            />
            <q-btn
              flat
              round
              color="grey"
              icon="fas fa-retweet"
              size="sm"
            />
            <q-btn
              @click="toggleLiked(qweet)"
              flat
              round
              :color="qweet.liked ? 'pink' : 'grey'"
              :icon="qweet.liked ? 'fas fa-heart': 'far fa-heart'"
              size="sm"
            />
            <q-btn
              @click="deleteTweet(qweet)"
              flat
              round
              color="grey"
              icon="fas fa-trash"
              size="sm"
            />
          </div>
        </q-item-section>

      </q-item>
      </transition-group>
    </q-list>
    </q-scroll-area>
  </q-page>
</template>

<script>
import { defineComponent } from 'vue';
import { formatDistance } from 'date-fns'
import db from 'src/boot/firebase'
import { collection,doc, query, orderBy, onSnapshot, addDoc, deleteDoc, updateDoc } from "firebase/firestore";

export default defineComponent({
  name: 'PageHome',
  data () {
    return{
      newQweetContent: '',
      qweets: [
        // {
        //   id: 'ID1',
        //   content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vitae cursus augue. Phasellus suscipit semper ante, non dictum justo venenatis sit amet. Suspendisse non nisi quis arcu tincidunt semper. ',
        //   date: 1634641094412,
        //   liked: false
        // },
        // {
        //   id: 'ID2',
        //   content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vitae cursus augue. Phasellus suscipit semper ante, non dictum justo venenatis sit amet. Suspendisse non nisi quis arcu tincidunt semper. Quisque vulputate suscipit augue, congue dapibus mauris sodales nec. Cras volutpat at lacus eget condimentum.',
        //   date: 1634641094412,
        //   liked: true
        // },
      ]
    }
  },
  methods:{
    relativeDate(value){
     return formatDistance(value, new Date());
    },
    async addNewTweet(){
      let newTweet = {
        content: this.newQweetContent,
        date: Date.now(),
        liked: false
      }
      //this.qweets.unshift(newTweet)
      // Add a new document with a generated id.
      const docRef = await addDoc(collection(db, "qweets"), newTweet)
      console.log("Document written with ID: ", docRef.id)
      this.newQweetContent = ''
    },
     async deleteTweet(qweet){
      //console.log('Delete tweet', item)
      //let dateToDelete = item.date
      //let index = this.qweets.findIndex(items => items.date === dateToDelete)
     // console.log('index: ' , index)
     // this.qweets.splice(index, 1)
       await deleteDoc(doc(db, 'qweets', qweet.id))
    },
    async toggleLiked(qweet){

      const likedRef = doc(db, "qweets", qweet.id);

// Set the "capital" field of the city 'DC'
      await updateDoc(likedRef, {
        liked: !qweet.liked
      });
    }
  },
  mounted() {
    const q = query(collection(db, 'qweets'), orderBy("date"));
    const unsubscribe = onSnapshot(q, (snapshot) => {
      snapshot.docChanges().forEach((change) => {
        let qweetChange = change.doc.data()
        qweetChange.id = change.doc.id
        if (change.type === 'added') {
          console.log('New tweet: ', qweetChange)
          this.qweets.unshift(qweetChange)
        }
        if (change.type === 'modified') {
          console.log('Modified tweet: ', qweetChange)
          let index = this.qweets.findIndex(tweet => tweet.id === qweetChange.id)
          Object.assign(this.qweets[index], qweetChange)
        }
        if (change.type === 'removed') {
          console.log('Removed tweet: ', qweetChange)
          let index = this.qweets.findIndex(tweet => tweet.id === qweetChange.id)
          this.qweets.splice(index, 1)
        }
      })
    });

  }
})
</script>

<style lang="sass">
.new-qweet
  textarea
      font-size: 19px
      line-height: 1.4 !important
.divider
      border-top: 1px solid
      border-bottom: 1px solid
      border-color: $grey-4
.qweet:not(:first-child)
  border-top: 1px solid rgba(0,0,0,0.12)
.qweet-content
  white-space: pre-line
.qweet-icons
  margin-left: -5px
</style>

**Comment by hawkeye64** [comment, marked as answer]
You can try the quasar CSS classes `ellipsis-2-lines` or `ellipsis-3-lines` depending on your needs. Also, play with CSS attribute `word-break`: https://developer.mozilla.org/en-US/docs/Web/CSS/word-break
